### PR TITLE
Explore: Integration test for running a query and saving it in query history

### LINKED ISSUE #45179: Explore: Integration test for running a query and saving it in query history

### PR DESCRIPTION
**Pull request by ifrost**
**What this PR does / why we need it**:

Integration test for Explore to cover a smoke test for query history.

Changes apart from adding `queryHistory.test.ts` with a smoke test:

* moved setup/interactions functions from Wrapper.test.tsx to separate `spec/helper` files
* `setupExplore()` is what we had before as `setup()` in `Wrapper.test.tsx` with small changes:
  * it allows optional skipping clearing local storage (required for the test)
  * it returns `unmount` (required for the test)
  * I've added `<input ../>` to mocked QueryEditor to allow changing the query

**Which issue(s) this PR fixes**:

Fixes #45179

**Special notes for your reviewer**: 

I think it can be improved further but I wanted to limit the scope. What we can do:
* Move Wrapper tests for URL parsing and split to a separate files in the `spec` folder
* Do not use explicit data source references (like Loki or Elastic) and use something generic instead

### CODE REVIEW

**Review by gelicia — APPROVED** [review]
Great cleanup and consolidation!

File: public/app/features/explore/spec/helper/setup.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+};
+
+export function setupExplore(options?: SetupOptions): {
+  datasources: { [name: string]: DataSourceApi };
+  store: EnhancedStore;
+  unmount: () => void;
+} {
+  // Clear this up otherwise it persists data source selection
+  // TODO: probably add test for that too
+  if (options?.clearLocalStorage !== false) {
```
[the remark this answers is not in the document]
Comment (ifrost) [reply public/app/features/explore/spec/helper/setup.tsx:37]: Good point! Added tests clean up in a9c564a0c693378649050009fa87576da252c94b

File: public/app/features/explore/spec/helper/interactions.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  const button = screen.getByTestId(selectors.components.RefreshPicker.runButtonV2);
+  userEvent.click(button);
+};
+
+export const openQueryHistory = async () => {
+  const button = screen.getByRole('button', { name: 'Rich history button' });
+  userEvent.click(button);
+  await waitFor(() => {
+    expect(screen.getByText('The history is local to your browser and is not shared with others.')).toBeInTheDocument();
+  });
```
Comment (Elfo404) [inline public/app/features/explore/spec/helper/interactions.ts:27]: you can just do

```suggestion
  expect(await screen.findByText('The history is local to your browser and is not shared with others.')).toBeInTheDocument();
```

(findBy* is a wrapper around waitFor and getBy*)

**Review by Elfo404 — APPROVED** [review]
Great stuff! 🚀  

left a couple of comment but none of them are blocking

File: public/app/features/explore/spec/helper/interactions.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  fireEvent.click(option);
+};
+
+export const inputQuery = (query: string) => {
+  const input = screen.getByRole('textbox', { name: 'query' });
+  userEvent.type(input, query);
+};
+
+export const runQuery = () => {
+  const button = screen.getByTestId(selectors.components.RefreshPicker.runButtonV2);
```
[the remark this answers is not in the document]
Comment (ifrost) [reply public/app/features/explore/spec/helper/interactions.ts:18]: Turns out the button didn't have an accessible name so I added it in e28d136eac0bfc7dd4bdade48b36cd35fae9d7ff. As a "side effect" we also get the tooltip on small screens:

Screen Shot 2022-02-23 at 16 24 39

**Pull request merged**